conan/api/subapi/cache.py

```diff
@@ -77,6 +80,69 @@ def check_integrity(self, package_list):
         checker = IntegrityChecker(cache)
         checker.check(package_list)
 
+    def sign(self, package_list):
+        """Sign packages with the package signing plugin"""
+        cache = PkgCache(self._conan_api.cache_folder, self._api_helpers.global_conf)
+        pkg_signer = PkgSignaturesPlugin(cache, self._conan_api.home_folder)
+        if not pkg_signer.is_configured:
+            raise ConanException(
+                "The package sign plugin is not configured. For more information on how to "
+                "configure it, please read the documentation at "
+                "https://docs.conan.io/2/reference/extensions/package_signing.html.")
+        app = ConanApp(self._conan_api)
+        preparator = PackagePreparator(app, self._api_helpers.global_conf)
+        preparator.prepare(package_list, [], force=True)
+
+        for rref, packages in package_list.items():
+            recipe_bundle = package_list.recipe_dict(rref)
+            if recipe_bundle:
```

Comment:
I think this should enforce a ``recipe_bundle`` to exist. If it doesn't exist it will skip it, doing nothing, giving a false sense of security "things are signed properly", but this is not true.